recipes/eigen/all/conanfile.py

```diff
@@ -0,0 +1,44 @@
+import os
+from conans import ConanFile, tools
+
+class eigenConan(ConanFile):
+    name = "eigen"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://eigen.tuxfamily.org"
+    description = "Eigen is a C++ template library for linear algebra: matrices, vectors, \
+                   numerical solvers, and related algorithms."
+    license = "MPL-2.0"
+    topics = ("eigen", "algebra", "linear-algebra", "vector", "numerical")
+    _source_subfolder = "_source_subfolder"
+    no_copy_source = True
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        #Get te extracted folder name. They allways have the format eigen-eigen-xxxxxx
+        listdir = os.listdir()
+        extracted_dir = [i for i in listdir if "eigen-eigen" in i][0]
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def package(self):
+        unsupported_folder = os.path.join(self.package_folder, "include", "eigen3", "unsupported", "Eigen")
+        eigen_folder = os.path.join(self.package_folder, "include", "eigen3", "Eigen")
+        self.copy("COPYING.*", dst="licenses", src=self._source_subfolder)
+        self.copy("*", dst=eigen_folder, src=os.path.join(self._source_subfolder, "Eigen"))
+        self.copy("*", dst=unsupported_folder, src=os.path.join(self._source_subfolder, "unsupported", "Eigen"))
+        self.copy("signature_of_eigen3_matrix_library", dst=os.path.join("include", "eigen3"), src=self._source_subfolder)
+        os.remove(os.path.join(eigen_folder, "CMakeLists.txt"))
+        os.remove(os.path.join(unsupported_folder, "CMakeLists.txt"))
+        os.remove(os.path.join(unsupported_folder, "CXX11", "CMakeLists.txt"))
+        os.remove(os.path.join(unsupported_folder, "CXX11", "src", "Tensor", "README.md"))
+        os.remove(os.path.join(unsupported_folder, "src", "EulerAngles", "CMakeLists.txt"))
+        os.rename(os.path.join(unsupported_folder, "src", "LevenbergMarquardt", "CopyrightMINPACK.txt"),
+                               os.path.join(self.package_folder, "licenses", "CopyrightMINPACK.txt"))
+
+    def package_id(self):
+        self.info.header_only()
+
+    def package_info(self):
+        self.cpp_info.name = "Eigen3"
+        self.cpp_info.includedirs = [os.path.join("include","eigen3")]
+        if tools.os_info.is_linux:
```

Comment:
I think it will be better for people cross-building to do this check like this:

```suggestion
        if self.settings.os == "Linux":
```
Adding the settings in the top of the recipe `settings = "os", "compiler", "arch", "build_type"`

Reply:
It is a header only library, are this settings needed?

Reply:
yes, I know. But you are making a OS check here, so better have the setting full settings for cross-building with other packages. AS you already have the `header_only()` in package info, it should be ok.